packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -660,6 +765,60 @@ const VirtualListBaseFactory = (type) => {
 			}
 		}
 
+		// For NewVirtualList
+		adjustVariableGridPosition () {
+			const
+				{dataSize, overhang, spacing, variableGridSizes} = this.props,
+				{firstIndex, numOfItems} = this.state,
+				{itemContainerRef, maxFirstIndex, variableGridPositions} = this,
+				lastIndex = firstIndex + numOfItems - 1,
+				numOfUpperLine = Math.floor(overhang / 2);
+
+			if (itemContainerRef.current) {
+				let index;
+
+				// Cache variable item positions
+				// and adjust variable item DOM element positions
+				for (index = firstIndex; index <= lastIndex; index++) {
+					const childNode = itemContainerRef.current.children[index % numOfItems];
+
+					if (!variableGridPositions[index] && variableGridSizes[index]) {
+						const position = index === 0 ? 0 : this.getVariableGridPosition(index - 1) + spacing;
+
+						variableGridPositions[index] = {position};
+					}
+
+					if (childNode && variableGridPositions[index]) {
+						childNode.style.transform = `translate3d(0, ${variableGridPositions[index].position}px, 0)`;
+					}
+				}
+
+				// Update threshold
+				this.threshold.min = firstIndex === 0 ? -Infinity : this.getVariableGridPosition(firstIndex + numOfUpperLine * 1);
+				this.threshold.max = lastIndex === maxFirstIndex ? Infinity : this.getVariableGridPosition(firstIndex + (numOfUpperLine + 1));
+
+				// Update scroll bounds
+				if (variableGridPositions.filter(Boolean).length === dataSize) { // all item sizes are known
+					this.scrollBounds.scrollHeight = variableGridSizes.reduce((acc, cur) => acc + cur, 0) + (dataSize - 1) * spacing;
```

Comment:
If a list scrolls from the starting point 0 as we assume, we're able to get the last position from the last entry of size's array.

Reply:
Let's use `getVariableGridBottomPosition` function to get the last position.

Reply:
We found there was a problem using`getVariableGridBottomPosition`. Items were not shown properly. Let's try to fix this issue later when enhancing performance.